web/src/user_status_ui.ts

```diff
@@ -74,45 +81,50 @@ export function submit_new_status() {
     });
 }
 
-export function update_button() {
+export function update_button(): void {
     const user_id = people.my_current_user_id();
     let old_status_text = user_status.get_status_text(user_id) ?? "";
     old_status_text = old_status_text.trim();
-    const old_emoji_info = user_status.get_status_emoji(user_id) || {};
-    const new_status_text = input_field().val().trim();
+    const old_emoji_info = user_status.get_status_emoji_info_to_compare(user_id);
+    let new_status_text = input_field().val();
+    assert(typeof new_status_text === "string");
```

Comment:
`.val()` return `string`, `string[]` and `number` type. `trim` is only for `string`

Reply:
This came up in another PR today, but maybe this should just be changed to `.text()`?

Reply:
I think `toString()` might be a better choice. [Comment](https://github.com/zulip/zulip/pull/29089#discussion_r1506010886) that I gave in that very PR explaining it.

Reply:
Sounds good!